File: Engine/Axis/Source/Engine/Utilities/TexCoords/AxTransformTexCoords.cpp
//---------------------------------------------------------------------------------------------
//	Axis 3D Engine Copyright (C) 2017 Palitri
//
//	This software is released under the MIT License http://palitri.com/axis/license/
//
//	For help and documentation, visit http://palitri.com
//---------------------------------------------------------------------------------------------

#include "AxTransformTexCoords.h"

#include "..\..\Primitive\AxVector2.h"

AxTransformTexCoords::AxTransformTexCoords(AxDeviceMesh &result, AxMatrix &transformation)
{
	int vertexCount = result.GetVertexCount();
	for (int i = 0; i < vertexCount; i++)
	{
		AxVector2 texCoords;
		result.GetVertexTexCoords(i, texCoords);
		AxVector2::Transform(texCoords, texCoords, transformation);
		result.SetVertexTexCoords(i, texCoords);
	}
}


AxTransformTexCoords::~AxTransformTexCoords(void)
{
}

File: Web/Engine/Utilities/TexCoords/AxTransformTexCoords.js
/**
 * Transforms the texture coordinates of a mesh
 * @param {AxDeviceMesh} result The mesh to have its texture coordinates transformed
 * @param {AxMatrix} transformation The transformation to apply to the texture coordinates of the mesh
 */
function AxTransformTexCoords(result, transformation)
{
    var vertexCount = result.GetVertexCount();
    for (var i = 0; i < vertexCount; i++)
    {
        var texCoords = new AxVector2();
        result.GetVertexTexCoords(i, texCoords);
        AxVector2.Transform(texCoords, texCoords, transformation);
        result.SetVertexTexCoords(i, texCoords);
    }
}
